### PR TITLE
py-keyring: fix installation on linux

### PR DESCRIPTION
**Pull request by manuelakuhn**
This depends on #24705

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-keyring/package.py [hunk 1]
```diff
@@ -20,4 +20,7 @@ class PyKeyring(PythonPackage):
     depends_on('py-setuptools', type='build')
     depends_on('py-setuptools-scm@3.4.1:+toml', type='build')
     depends_on('py-importlib-metadata@3.6:', type=('build', 'run'))
-    # TODO: additional dependencies required for Windows/Linux
+    depends_on('py-secretstorage@3.2:', type='run', when='platform=linux')
+    depends_on('py-jeepney@0.4.1:', type='run', when='platform=linux')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-keyring/package.py]:
```suggestion
    depends_on('py-jeepney@0.4.2:', type=('build', 'run'), when='platform=linux')
```

File: var/spack/repos/builtin/packages/py-keyring/package.py [hunk 2]
```diff
@@ -20,4 +20,7 @@ class PyKeyring(PythonPackage):
     depends_on('py-setuptools', type='build')
     depends_on('py-setuptools-scm@3.4.1:+toml', type='build')
     depends_on('py-importlib-metadata@3.6:', type=('build', 'run'))
-    # TODO: additional dependencies required for Windows/Linux
+    depends_on('py-secretstorage@3.2:', type='run', when='platform=linux')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-keyring/package.py]:
```suggestion
    depends_on('py-secretstorage@3.2:', type=('build', 'run'), when='platform=linux')
```

File: var/spack/repos/builtin/packages/py-keyring/package.py [hunk 3]
```diff
@@ -20,4 +20,7 @@ class PyKeyring(PythonPackage):
...
+    depends_on('py-jeepney@0.4.1:', type='run', when='platform=linux')
+
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-keyring/package.py:25]: Can you also add a build dep on `py-setuptools-scm@3.4.1:+toml`?

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-keyring/package.py (hunk 3 above)
Comment (manuelakuhn) [reply var/spack/repos/builtin/packages/py-keyring/package.py:25]: Sorry, I am not really sure what you mean, there is already a build dependency on `py-setuptools-scm@3.4.1:+toml` in line 21.

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-keyring/package.py:25]: Oops, didn't see that. Thanks!

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**